File: 24-SwapCase/24.hs
--Given a string, swap the cases of each letter in the string
import Data.Char

swapCase :: String -> String
swapCase = map (\x -> if isUpper x then toLower x else toUpper x) 

main = do
  print $ swapCase "Hello World" == "hELLO wORLD"
  print $ swapCase "Wow. WOW!"   == "wOW. wow!"

File: 24-SwapCase/24.js
//Given a string, swap the cases of each letter in the string

function SwapCase(str) { 
  return str.split('').reduce(function(acc,x){
      return acc + ((x === x.toUpperCase()) ? x.toLowerCase() : x.toUpperCase());
    },'');  
}

console.log(SwapCase("Hello World") === "hELLO wORLD");
console.log(SwapCase("Wow. WOW!") === "wOW. wow!");
